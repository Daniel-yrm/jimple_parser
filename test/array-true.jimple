public final class MainKt extends java.lang.Object
{

    public static final void main()
    {
        java.lang.AssertionError $r1;
        int[] r0;
        int $i0, $i1, $i2, i3, i4, i5, i6;
        boolean $z0, $z1;

        r0 = newarray (int)[20];

        i3 = 0;

     label1:
        if i3 >= 20 goto label2;

        i4 = i3;

        i3 = i3 + 1;

        $i2 = 20 - i4;

        r0[i4] = $i2;

        goto label1;

     label2:
        i5 = 0;

     label3:
        if i5 >= 20 goto label6;

        i6 = i5;

        i5 = i5 + 1;

        $i1 = r0[i6];

        $i0 = 20 - i6;

        if $i1 != $i0 goto label4;

        $z1 = 1;

        goto label5;

     label4:
        $z1 = 0;

     label5:
        $z0 = <kotlin._Assertions: boolean ENABLED>;

        if $z0 == 0 goto label3;

        if $z1 != 0 goto label3;

        $r1 = new java.lang.AssertionError;

        specialinvoke $r1.<java.lang.AssertionError: void <init>(java.lang.Object)>("Assertion failed");

        throw $r1;

     label6:
        return;
    }

    public static void main(java.lang.String[])
    {
        java.lang.String[] r0;

        r0 := @parameter0: java.lang.String[];

        staticinvoke <MainKt: void main()>();

        return;
    }
}
