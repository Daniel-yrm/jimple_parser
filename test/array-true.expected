{"extends":"java.lang.Object","object":"Class","name":"MainKt","modifiers":["public","final"],"content":[{"object":"Method","name":"main","modifiers":["public","static","final"],"content":[{"object":"Variable","name":"$r1","type":{"identifier":"java.lang.AssertionError","dimensions":0}},{"object":"Variable","name":"r0","type":{"identifier":"int","dimensions":1}},{"object":"Variable","name":"$i0","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"$i1","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"$i2","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"i3","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"i4","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"i5","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"i6","type":{"identifier":"int","dimensions":0}},{"object":"Variable","name":"$z0","type":{"identifier":"boolean","dimensions":0}},{"object":"Variable","name":"$z1","type":{"identifier":"boolean","dimensions":0}},{"object":"SetVariable","name":"r0","value":{"expr_type":"newarray","type":{"identifier":"int","dimensions":0},"size":{"expr_type":"constant","value":"20"}}},{"object":"SetVariable","name":"i3","value":{"expr_type":"constant","value":"0"}},{"label_id":"label1","object":"Label","content":[]},{"goto":"label2","object":"If","expression":{"rhs":{"expr_type":"constant","value":"20"},"operator":">=","expr_type":"binop","lhs":{"expr_type":"symbol","value":"i3"}}},{"object":"SetVariable","name":"i4","value":{"expr_type":"symbol","value":"i3"}},{"object":"SetVariable","name":"i3","value":{"rhs":{"expr_type":"constant","value":"1"},"operator":"+","expr_type":"binop","lhs":{"expr_type":"symbol","value":"i3"}}},{"object":"SetVariable","name":"$i2","value":{"rhs":{"expr_type":"symbol","value":"i4"},"operator":"-","expr_type":"binop","lhs":{"expr_type":"constant","value":"20"}}},{"pos":{"expr_type":"symbol","value":"i4"},"object":"SetVariableDeref","name":"r0","value":{"expr_type":"symbol","value":"$i2"}},{"goto":"label1","object":"Goto"},{"label_id":"label2","object":"Label","content":[]},{"object":"SetVariable","name":"i5","value":{"expr_type":"constant","value":"0"}},{"label_id":"label3","object":"Label","content":[]},{"goto":"label6","object":"If","expression":{"rhs":{"expr_type":"constant","value":"20"},"operator":">=","expr_type":"binop","lhs":{"expr_type":"symbol","value":"i5"}}},{"object":"SetVariable","name":"i6","value":{"expr_type":"symbol","value":"i5"}},{"object":"SetVariable","name":"i5","value":{"rhs":{"expr_type":"constant","value":"1"},"operator":"+","expr_type":"binop","lhs":{"expr_type":"symbol","value":"i5"}}},{"object":"SetVariable","name":"$i1","value":{"expr_type":"deref","index":{"expr_type":"symbol","value":"i6"},"base":{"expr_type":"symbol","value":"r0"}}},{"object":"SetVariable","name":"$i0","value":{"rhs":{"expr_type":"symbol","value":"i6"},"operator":"-","expr_type":"binop","lhs":{"expr_type":"constant","value":"20"}}},{"goto":"label4","object":"If","expression":{"rhs":{"expr_type":"symbol","value":"$i0"},"operator":"notequal","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$i1"}}},{"object":"SetVariable","name":"$z1","value":{"expr_type":"constant","value":"1"}},{"goto":"label5","object":"Goto"},{"label_id":"label4","object":"Label","content":[]},{"object":"SetVariable","name":"$z1","value":{"expr_type":"constant","value":"0"}},{"label_id":"label5","object":"Label","content":[]},{"object":"SetVariable","name":"$z0","value":{"expr_type":"field_access","from":"kotlin._Assertions","type":{"identifier":"boolean","dimensions":0},"field":"ENABLED"}},{"goto":"label3","object":"If","expression":{"rhs":{"expr_type":"constant","value":"0"},"operator":"==","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$z0"}}},{"goto":"label3","object":"If","expression":{"rhs":{"expr_type":"constant","value":"0"},"operator":"notequal","expr_type":"binop","lhs":{"expr_type":"symbol","value":"$z1"}}},{"object":"SetVariable","name":"$r1","value":{"expr_type":"new","type":{"identifier":"java.lang.AssertionError","dimensions":0}}},{"base_class":"java.lang.AssertionError","parameter_type":[{"identifier":"java.lang.Object","dimensions":0}],"object":"SpecialInvoke","method":"<init>","variable":"$r1","parameters":[]},{"expr":{"expr_type":"symbol","value":"$r1"},"object":"Throw"},{"label_id":"label6","object":"Label","content":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[]},{"object":"Method","name":"main","modifiers":["public","static"],"content":[{"object":"Variable","name":"r0","type":{"identifier":"java.lang.String","dimensions":1}},{"object":"SetVariable","name":"r0","value":{"to":{"identifier":"java.lang.String","dimensions":1},"expr_type":"cast","from":{"expr_type":"symbol","value":"@parameter0"}}},{"base_class":"MainKt","object":"StaticInvoke","method":"main","parameters":[]},{"object":"Return"}],"type":{"identifier":"void","dimensions":0},"parameters":[{"identifier":"java.lang.String","dimensions":1}]}]}